---
output: github_document
---

<!-- README.md is generated from README.Rmd. Please edit that file -->

```{r, include = FALSE}
knitr::opts_chunk$set(
  collapse = TRUE,
  comment = "#>",
  fig.path = "man/figures/README-",
  out.width = "100%"
)
```

# crosshap

<!-- badges: start -->
<!-- badges: end -->

LD-based local haplotype analysis and visualization tool

## What does it do?

Given a genomic variant data for a region of interest, crosshap performs
LD-based local haplotyping before providing a range of visualization options 
to thoroughly examine the core patterns of co-inheritance present in light of 
optional phenotype/metadata.

## Crosshap VIZ

## Why would I use it?

crosshap was originally designed to explore local haplotype patterns that
may underlie phenotypic variability in quantitative trait locus (QTL) regions.
It is ideally suited to complement and follow-up GWAS results (takes same 
inputs). crosshap equips users with the tools to explain why a region reported a 
GWAS hit, what variants are causal candidates, what populations are they 
present/absent in, and what the features are of those populations.

Alternatively, crosshap can simply be a tool to identify patterns of linkage
among local variants, and to classify individuals based on shared haplotypes.

Note: crosshap is designed for in-depth, user-driven analysis of inheritance 
patterns in specific regions of interest, not genome-wide scans.

## Installation

You can install the development version of crosshap from [GitHub](https://github.com/) with:

``` r
# install.packages("devtools")
devtools::install_github("JacobIMarsh/crosshap")
```

## Usage

Workflow is >>> 








What is special about using `README.Rmd` instead of just `README.md`? You can include R chunks like so:

```{r cars}
summary(cars)
```

You'll still need to render `README.Rmd` regularly, to keep `README.md` up-to-date. `devtools::build_readme()` is handy for this. You could also use GitHub Actions to re-render `README.Rmd` every time you push. An example workflow can be found here: <https://github.com/r-lib/actions/tree/v1/examples>.

You can also embed plots, for example:

```{r pressure, echo = FALSE}
plot(pressure)
```

In that case, don't forget to commit and push the resulting figure files, so they display on GitHub and CRAN.
