<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Visualize haplotypes — crosshap_viz • crosshap</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Visualize haplotypes — crosshap_viz"><meta name="description" content="crosshap_viz() builds five individual plots using various elements of a
HapObject created by run_haplotyping(). The central dotplot displays
relationship between clusters of linked SNPs (Marker Groups), and distinct
haplotypes present within the population. Vertical plots (top/bottom)
visualize individuals and populations, grouped by haplotype. Horizontal plots
(left/right) visualize SNP information, grouped by Marker Group cluster."><meta property="og:description" content="crosshap_viz() builds five individual plots using various elements of a
HapObject created by run_haplotyping(). The central dotplot displays
relationship between clusters of linked SNPs (Marker Groups), and distinct
haplotypes present within the population. Vertical plots (top/bottom)
visualize individuals and populations, grouped by haplotype. Horizontal plots
(left/right) visualize SNP information, grouped by Marker Group cluster."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a></li>
    <li><a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jacobimarsh/crosshap/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Visualize haplotypes</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacobimarsh/crosshap/blob/main/R/crosshapviz_wrapper.R" class="external-link"><code>R/crosshapviz_wrapper.R</code></a></small>
      <div class="d-none name"><code>crosshap_viz.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>crosshap_viz() builds five individual plots using various elements of a
HapObject created by run_haplotyping(). The central dotplot displays
relationship between clusters of linked SNPs (Marker Groups), and distinct
haplotypes present within the population. Vertical plots (top/bottom)
visualize individuals and populations, grouped by haplotype. Horizontal plots
(left/right) visualize SNP information, grouped by Marker Group cluster.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">crosshap_viz</span><span class="op">(</span></span>
<span>  <span class="va">HapObject</span>,</span>
<span>  <span class="va">epsilon</span>,</span>
<span>  plot_left <span class="op">=</span> <span class="st">"allele"</span>,</span>
<span>  plot_right <span class="op">=</span> <span class="st">"pheno"</span>,</span>
<span>  hide_labels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  isolate_group <span class="op">=</span> <span class="cn">NA</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-hapobject">HapObject<a class="anchor" aria-label="anchor" href="#arg-hapobject"></a></dt>
<dd><p>Haplotype object created by run_haplotyping().</p></dd>


<dt id="arg-epsilon">epsilon<a class="anchor" aria-label="anchor" href="#arg-epsilon"></a></dt>
<dd><p>Epsilon to visualize haplotyping results for.</p></dd>


<dt id="arg-plot-left">plot_left<a class="anchor" aria-label="anchor" href="#arg-plot-left"></a></dt>
<dd><p>When plot_left = "allele", SNP allele frequency information
is displayed, when plot_left = "pos", SNP position information is displayed.</p></dd>


<dt id="arg-plot-right">plot_right<a class="anchor" aria-label="anchor" href="#arg-plot-right"></a></dt>
<dd><p>When plot_right = "pheno", phenotype associations for SNPs
are displayed, when plot_right = "cluster", internal marker group linkage is
displayed.</p></dd>


<dt id="arg-hide-labels">hide_labels<a class="anchor" aria-label="anchor" href="#arg-hide-labels"></a></dt>
<dd><p>When TRUE, legends from plots are hidden.</p></dd>


<dt id="arg-isolate-group">isolate_group<a class="anchor" aria-label="anchor" href="#arg-isolate-group"></a></dt>
<dd><p>If one or more Metadata groups are provided, all other
Metadata groups will be masked from the plot. NOTE: it does change the
summary tables or marker group phenotype scores.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A patchwork object.</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jacob Marsh, Brady Johnston, Jakob Petereit.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

