<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Package index â€¢ crosshap</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Package index"></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a></li>
    <li><a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jacobimarsh/crosshap/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Package index</h1>
    </div>

    <div class="section level2">
      <h2 id="core-analysis-functions">Core analysis functions<a class="anchor" aria-label="anchor" href="#core-analysis-functions"></a></h2>

      <div class="section-desc"><p>Analysis and visualization functions</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="run_haplotyping.html">run_haplotyping()</a></code>

        </dt>
        <dd>Cluster SNPs and identify haplotypes</dd>
      </dl><dl><dt>

          <code><a href="clustree_viz.html">clustree_viz()</a></code>

        </dt>
        <dd>Clustering tree</dd>
      </dl><dl><dt>

          <code><a href="crosshap_viz.html">crosshap_viz()</a></code>

        </dt>
        <dd>Visualize haplotypes</dd>
      </dl><dl><dt>

          <code><a href="prepare_hap_umap.html">prepare_hap_umap()</a></code>

        </dt>
        <dd>UMAP haplotype visualization helper</dd>
      </dl></div><div class="section level2">
      <h2 id="helper-functions">Helper functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h2>

      <div class="section-desc"><p>Functions to read in and format data for haplotyping</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="read_LD.html">read_LD()</a></code>

        </dt>
        <dd>Read LD correlation matrix to tibble</dd>
      </dl><dl><dt>

          <code><a href="read_metadata.html">read_metadata()</a></code>

        </dt>
        <dd>Read metadata to tibble</dd>
      </dl><dl><dt>

          <code><a href="read_vcf.html">read_vcf()</a></code>

        </dt>
        <dd>Read VCF to tibble</dd>
      </dl><dl><dt>

          <code><a href="read_pheno.html">read_pheno()</a></code>

        </dt>
        <dd>Read phenotype data to tibble</dd>
      </dl></div><div class="section level2">
      <h2 id="individual-crosshap-plots">Individual crosshap plots<a class="anchor" aria-label="anchor" href="#individual-crosshap-plots"></a></h2>

      <div class="section-desc"><p>Functions for building standalone plots that are stitched together by crosshap_viz()</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="build_bot_halfeyeplot.html">build_bot_halfeyeplot()</a></code>

        </dt>
        <dd>Bot hap-pheno raincloud plot</dd>
      </dl><dl><dt>

          <code><a href="build_left_alleleplot.html">build_left_alleleplot()</a></code>

        </dt>
        <dd>Left SNP-allele plot</dd>
      </dl><dl><dt>

          <code><a href="build_left_posplot.html">build_left_posplot()</a></code>

        </dt>
        <dd>Left SNP-position plot</dd>
      </dl><dl><dt>

          <code><a href="build_mid_dotplot.html">build_mid_dotplot()</a></code>

        </dt>
        <dd>Middle MG/hap dot plot</dd>
      </dl><dl><dt>

          <code><a href="build_right_clusterplot.html">build_right_clusterplot()</a></code>

        </dt>
        <dd>Right intra-cluster linkage plot</dd>
      </dl><dl><dt>

          <code><a href="build_right_phenoplot.html">build_right_phenoplot()</a></code>

        </dt>
        <dd>Right SNP-pheno phenoplot</dd>
      </dl><dl><dt>

          <code><a href="build_top_metaplot.html">build_top_metaplot()</a></code>

        </dt>
        <dd>Top metadata-hap bar plot</dd>
      </dl><dl><dt>

          <code><a href="build_summary_tables.html">build_summary_tables()</a></code>

        </dt>
        <dd>Hap/MG summary tables</dd>
      </dl></div><div class="section level2">
      <h2 id="example-data">Example data<a class="anchor" aria-label="anchor" href="#example-data"></a></h2>

      <div class="section-desc"><p>Example inputs for haplotyping and results for vignettes</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="LD.html">LD</a></code>

        </dt>
        <dd>Example LD matrix</dd>
      </dl><dl><dt>

          <code><a href="vcf.html">vcf</a></code>

        </dt>
        <dd>Example VCF</dd>
      </dl><dl><dt>

          <code><a href="metadata.html">metadata</a></code>

        </dt>
        <dd>Example Domestication metadata</dd>
      </dl><dl><dt>

          <code><a href="pheno.html">pheno</a></code>

        </dt>
        <dd>Example phenotype data</dd>
      </dl><dl><dt>

          <code><a href="HapObject.html">HapObject</a></code>

        </dt>
        <dd>Example Haplotype object</dd>
      </dl></div><div class="section level2">
      <h2 id="internal-functions">Internal functions<a class="anchor" aria-label="anchor" href="#internal-functions"></a></h2>

      <div class="section-desc"><p>Functions called internally by crosshap to perform certain tasks</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="arith_mode.html">arith_mode()</a></code>

        </dt>
        <dd>Mode utility function</dd>
      </dl><dl><dt>

          <code><a href="mean_na.rm.html">mean_na.rm()</a></code>

        </dt>
        <dd>Mean utility function</dd>
      </dl><dl><dt>

          <code><a href="pseudo_haps.html">pseudo_haps()</a></code>

        </dt>
        <dd>Identify haplotypes from clustered SNPs</dd>
      </dl><dl><dt>

          <code><a href="tagphenos.html">tagphenos()</a></code>

        </dt>
        <dd>Calculate SNP phenotypic associations</dd>
      </dl></div><div class="section level2">
      <h2 id="deprecated-functions">Deprecated functions<a class="anchor" aria-label="anchor" href="#deprecated-functions"></a></h2>

      <div class="section-desc"><p>Functions from previous versions that are not actively tested and supported</p></div>


    </div><div class="section level2">




      <dl><dt>

          <code><a href="run_hdbscan_haplotyping.html">run_hdbscan_haplotyping()</a></code>

        </dt>
        <dd>Cluster SNPs with HDBSCAN and identify haplotypes</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jacob Marsh, Brady Johnston, Jakob Petereit.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

