<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Cluster SNPs with HDBSCAN and identify haplotypes — run_hdbscan_haplotyping • crosshap</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Cluster SNPs with HDBSCAN and identify haplotypes — run_hdbscan_haplotyping"><meta name="description" content="run_hdbscan_haplotyping() performs HDBSCAN clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be visualized with reference
to phenotype and metadata using crosshap_viz() (set epsilon to 1 as a dummy value)."><meta property="og:description" content="run_hdbscan_haplotyping() performs HDBSCAN clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be visualized with reference
to phenotype and metadata using crosshap_viz() (set epsilon to 1 as a dummy value)."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles"><li><a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a></li>
    <li><a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a></li>
    <li><a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a></li>
  </ul></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/jacobimarsh/crosshap/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Cluster SNPs with HDBSCAN and identify haplotypes</h1>
      <small class="dont-index">Source: <a href="https://github.com/jacobimarsh/crosshap/blob/main/R/run_hdbscan_haplotyping.R" class="external-link"><code>R/run_hdbscan_haplotyping.R</code></a></small>
      <div class="d-none name"><code>run_hdbscan_haplotyping.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>run_hdbscan_haplotyping() performs HDBSCAN clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be visualized with reference
to phenotype and metadata using crosshap_viz() (set epsilon to 1 as a dummy value).</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_hdbscan_haplotyping</span><span class="op">(</span></span>
<span>  <span class="va">vcf</span>,</span>
<span>  <span class="va">LD</span>,</span>
<span>  <span class="va">pheno</span>,</span>
<span>  <span class="va">MGmin</span>,</span>
<span>  minHap <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  hetmiss_as <span class="op">=</span> <span class="st">"allele"</span>,</span>
<span>  metadata <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  keep_outliers <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-vcf">vcf<a class="anchor" aria-label="anchor" href="#arg-vcf"></a></dt>
<dd><p>Input VCF for region of interest.</p></dd>


<dt id="arg-ld">LD<a class="anchor" aria-label="anchor" href="#arg-ld"></a></dt>
<dd><p>Pairwise correlation matrix of SNPs in region (e.g. from PLINK).</p></dd>


<dt id="arg-pheno">pheno<a class="anchor" aria-label="anchor" href="#arg-pheno"></a></dt>
<dd><p>Input numeric phenotype data for each individual.</p></dd>


<dt id="arg-mgmin">MGmin<a class="anchor" aria-label="anchor" href="#arg-mgmin"></a></dt>
<dd><p>Minimum SNPs in marker groups, MinPts parameter for DBscan.</p></dd>


<dt id="arg-minhap">minHap<a class="anchor" aria-label="anchor" href="#arg-minhap"></a></dt>
<dd><p>Minimum nIndividuals in a haplotype combination.</p></dd>


<dt id="arg-hetmiss-as">hetmiss_as<a class="anchor" aria-label="anchor" href="#arg-hetmiss-as"></a></dt>
<dd><p>If hetmiss_as = "allele", heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = "miss", the site is recoded as missing.</p></dd>


<dt id="arg-metadata">metadata<a class="anchor" aria-label="anchor" href="#arg-metadata"></a></dt>
<dd><p>Metadata input (optional).</p></dd>


<dt id="arg-keep-outliers">keep_outliers<a class="anchor" aria-label="anchor" href="#arg-keep-outliers"></a></dt>
<dd><p>When FALSE, marker group smoothing is performed to
remove outliers.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A comprehensive haplotyping S3 object (HapObject) for each provided
epsilon value, needed for clustree_viz() and crosshap_viz().</p>
    </div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Jacob Marsh, Brady Johnston, Jakob Petereit.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

