<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>Haplotype clustering and visualization â€¢ crosshap</title>
<!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous">
<script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="bootstrap-toc.css">
<script src="bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="pkgdown.css" rel="stylesheet">
<script src="pkgdown.js"></script><meta property="og:title" content="Haplotype clustering and visualization">
<meta property="og:description" content="More about what it does (maybe more than one line)
    Use four spaces when indenting paragraphs within the Description.">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body data-spy="scroll" data-target="#toc">
    

    <div class="container template-home">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="index.html">crosshap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="articles/Getting_started.html">Getting_started.Rmd</a>
    </li>
    <li>
      <a href="articles/talkin_cars.html">talkin_cars</a>
    </li>
  </ul>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      

      </header><div class="row">
  <div class="contents col-md-9">
<div class="section level1">
<div class="page-header"><h1 id="crosshap">crosshap<a class="anchor" aria-label="anchor" href="#crosshap"></a>
</h1></div>
<!-- badges: start -->
<!-- badges: end -->
<p>LD-based local haplotype analysis and visualization tool</p>
<div class="section level2">
<h2 id="what-does-it-do">What does it do?<a class="anchor" aria-label="anchor" href="#what-does-it-do"></a>
</h2>
<p>Given a genomic variant data for a region of interest, crosshap performs LD-based local haplotyping. Tightly linked variants are clustered into Marker Groups (MGs), and individuals are grouped into local haplotypes by shared allelic combinations of MGs. Following this, crosshap provides a range of visualization options to examine relevant characteristics of the linked Marker Groups and local haplotypes.</p>
</div>
<div class="section level2">
<h2 id="why-would-i-use-it">Why would I use it?<a class="anchor" aria-label="anchor" href="#why-would-i-use-it"></a>
</h2>
<p>crosshap was originally designed to explore local haplotype patterns that may underlie phenotypic variability in quantitative trait locus (QTL) regions. It is ideally suited to complement and follow-up GWAS results (takes same inputs). crosshap equips users with the tools to explain why a region reported a GWAS hit, what variants are causal candidates, what populations are they present/absent in, and what the features are of those populations.</p>
<p>Alternatively, crosshap can simply be a tool to identify patterns of linkage among local variants, and to classify individuals based on shared haplotypes.</p>
<p>Note: crosshap is designed for in-depth, user-driven analysis of inheritance patterns in specific regions of interest, not genome-wide scans.</p>
<div class="figure">
<img src="https://github.com/JacobIMarsh/crosshap/blob/main/images/crosshap_jpg.jpg?raw=true" alt=""><p class="caption">alt text</p>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>You can install the development version of crosshap from <a href="https://github.com/" class="external-link">GitHub</a> with:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># install.packages("devtools")</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"JacobIMarsh/crosshap"</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="usage">Usage<a class="anchor" aria-label="anchor" href="#usage"></a>
</h2>
<p>In short, a typical crosshap analysis workflow involves the following steps. For a detailed explanation and walk through, see <code>Getting_started.Rmd</code> (coming soon!).</p>
<ol start="0" style="list-style-type: decimal">
<li>Read in raw inputs</li>
</ol>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/read_vcf.html">read_vcf</a></span><span class="op">(</span><span class="va">region.vcf</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/read_LD.html">read_LD</a></span><span class="op">(</span><span class="va">plink.ld</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/read_metadata.html">read_metadata</a></span><span class="op">(</span><span class="va">metadata.txt</span><span class="op">)</span></span>
<span><span class="fu"><a href="reference/read_pheno.html">read_pheno</a></span><span class="op">(</span><span class="va">pheno.txt</span><span class="op">)</span></span></code></pre></div>
<ol style="list-style-type: decimal">
<li>Run local haplotyping at a range of epsilon values</li>
</ol>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/run_haplotyping.html">run_haplotyping</a></span><span class="op">(</span><span class="va">vcf</span>, <span class="va">LD</span>, <span class="va">metadata</span>, <span class="va">pheno</span><span class="op">)</span></span></code></pre></div>
<ol start="2" style="list-style-type: decimal">
<li>Build clustering tree to optimize epsilon value</li>
</ol>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/run_clustree.html">run_clustree</a></span><span class="op">(</span><span class="va">pheno</span><span class="op">)</span></span></code></pre></div>
<ol start="3" style="list-style-type: decimal">
<li>Visualize local haplotypes and Marker Groups</li>
</ol>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="reference/crosshap_viz.html">crosshap_viz</a></span><span class="op">(</span><span class="va">HapObject</span><span class="op">)</span></span></code></pre></div>
<p>From here you can examine haplotype and Marker Group features from the visualization, and export relevant information from the haplotype object.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">HapObject</span><span class="op">$</span><span class="va">Indfile</span></span>
<span><span class="va">HapObject</span><span class="op">$</span><span class="va">Varfile</span></span>
<span><span class="va">HapObject</span><span class="op">$</span><span class="va">Hapfile</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="contact">Contact<a class="anchor" aria-label="anchor" href="#contact"></a>
</h2>
<p>For technical queries feel free to contact me: <a href="mailto:jacob.marsh@research.uwa.edu.au" class="email">jacob.marsh@research.uwa.edu.au</a>. Please contact Prof.Â David Edwards for all other queries: <a href="mailto:dave.edwards@uwa.edu.au" class="email">dave.edwards@uwa.edu.au</a> .</p>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li>What license is it under?</li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing crosshap</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jacob Marsh <br><small class="roles"> Author, maintainer </small>  </li>
</ul>
</div>



  </div>
</div>


      <footer><div class="copyright">
  <p></p>
<p>Developed by Jacob Marsh.</p>
</div>

<div class="pkgdown">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer>
</div>

  


  

  </body>
</html>
