<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1.0"><title>Cluster SNPs and identify haplotypes — run_haplotyping • crosshap</title><!-- jquery --><script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script><!-- Bootstrap --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/css/bootstrap.min.css" integrity="sha256-bZLfwXAP04zRMK2BjiO8iu9pf4FbLqX6zitd+tIvLhE=" crossorigin="anonymous"><script src="https://cdnjs.cloudflare.com/ajax/libs/twitter-bootstrap/3.4.1/js/bootstrap.min.js" integrity="sha256-nuL8/2cJ5NDSSwnKD8VqreErSWHtnEP9E7AySL+1ev4=" crossorigin="anonymous"></script><!-- bootstrap-toc --><link rel="stylesheet" href="../bootstrap-toc.css"><script src="../bootstrap-toc.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet"><script src="../pkgdown.js"></script><meta property="og:title" content="Cluster SNPs and identify haplotypes — run_haplotyping"><meta property="og:description" content="run_haplotyping() performs density-based clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be used as input to
build clustering tree for epsilon optimization using run_clustree(), and
can be visualized with reference to phenotype and metadata using
crosshap_viz()."><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body data-spy="scroll" data-target="#toc">
    

    <div class="container template-reference-topic">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar" aria-expanded="false">
        <span class="sr-only">Toggle navigation</span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">crosshap</a>
        <span class="version label label-default" data-toggle="tooltip" data-placement="bottom" title="">0.1.0</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav"><li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu"><li>
      <a href="../articles/Getting_started.html">Getting_started.Rmd</a>
    </li>
    <li>
      <a href="../articles/talkin_cars.html">talkin_cars</a>
    </li>
  </ul></li>
      </ul><ul class="nav navbar-nav navbar-right"></ul></div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

      

      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header">
    <h1>Cluster SNPs and identify haplotypes</h1>
    
    <div class="hidden name"><code>run_haplotyping.Rd</code></div>
    </div>

    <div class="ref-description">
    <p>run_haplotyping() performs density-based clustering of SNPs in region of
interest to identify marker groups. Individuals are classified by haplotype
combination based on shared combinations of marker group alleles. Returns a
comprehensive haplotyping object (HapObject), which can be used as input to
build clustering tree for epsilon optimization using run_clustree(), and
can be visualized with reference to phenotype and metadata using
crosshap_viz().</p>
    </div>

    <div id="ref-usage">
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">run_haplotyping</span><span class="op">(</span></span>
<span>  <span class="va">vcf</span>,</span>
<span>  <span class="va">LD</span>,</span>
<span>  <span class="va">pheno</span>,</span>
<span>  metadata <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  epsilon <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>, <span class="fl">0.4</span>, <span class="fl">0.6</span>, <span class="fl">0.8</span>, <span class="fl">1</span><span class="op">)</span>,</span>
<span>  MGmin <span class="op">=</span> <span class="fl">30</span>,</span>
<span>  minHap <span class="op">=</span> <span class="fl">9</span>,</span>
<span>  hetmiss_as <span class="op">=</span> <span class="st">"allele"</span>,</span>
<span>  het_as <span class="op">=</span> <span class="st">"alt"</span>,</span>
<span>  keep_outliers <span class="op">=</span> <span class="cn">F</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div id="arguments">
    <h2>Arguments</h2>
    <dl><dt>vcf</dt>
<dd><p>Input VCF for region of interest.</p></dd>


<dt>LD</dt>
<dd><p>Pairwise correlation matrix of SNPs in region (e.g. from PLINK).</p></dd>


<dt>pheno</dt>
<dd><p>Input numeric phenotype data for each individual.</p></dd>


<dt>metadata</dt>
<dd><p>Metadata input (optional).</p></dd>


<dt>epsilon</dt>
<dd><p>Epsilon values for clustering SNPs with DBscan.</p></dd>


<dt>MGmin</dt>
<dd><p>Minimum SNPs in marker groups, MinPts parameter for DBscan.</p></dd>


<dt>minHap</dt>
<dd><p>Minimum nIndividuals in a haplotype combination.</p></dd>


<dt>hetmiss_as</dt>
<dd><p>If hetmiss_as = "allele", heterozygous-missing SNPs './N'
are recoded as 'N/N', if hetmiss_as = "miss", the site is recoded as missing.</p></dd>


<dt>het_as</dt>
<dd><p>If het_as = "alt", heterozygous SNPs are recoded 'REF/ALT' are
recoded as 'ALT/ALT' to reduce number of unique haplotypes, if het_as =
"het", they are kept as 'REF/ALT'.</p></dd>


<dt>keep_outliers</dt>
<dd><p>When FALSE, marker group smoothing is performed to
remove outliers.</p></dd>

</dl></div>
    <div id="value">
    <h2>Value</h2>
    

<p>A comprehensive haplotyping S3 object (HapObject) for each provided
epsilon value, needed for run_clustree() and crosshap_viz().</p>
    </div>

    <div id="ref-examples">
    <h2>Examples</h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span></span></span>
<span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span></span></span>
<span class="r-in"><span>     <span class="fu">run_haplotyping</span><span class="op">(</span><span class="va">vcf</span>, <span class="va">LD</span>, <span class="va">pheno</span>, <span class="va">metadata</span>, MGmin <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </div>
  <div class="col-md-3 hidden-xs hidden-sm" id="pkgdown-sidebar">
    <nav id="toc" data-toggle="toc" class="sticky-top"><h2 data-toc-skip>Contents</h2>
    </nav></div>
</div>


      <footer><div class="copyright">
  <p></p><p>Developed by Jacob Marsh.</p>
</div>

<div class="pkgdown">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

      </footer></div>

  


  

  </body></html>

