<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="crosshap">
<title>Delimiting a region of interest • crosshap</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Delimiting a region of interest">
<meta property="og:description" content="crosshap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a>
    <a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a>
    <a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Delimiting a region of interest</h1>
                        <h4 data-toc-skip class="author">Jacob
Marsh</h4>
            
            <h4 data-toc-skip class="date">2023-03-08</h4>
      
      
      <div class="d-none name"><code>Delimiting_region.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Local haplotyping first and foremost requires the specification of a
genomic region to analyse. The fundamental aim of delimiting a genomic
region to haplotype with crosshap should be to include the minimum range
of sites that sufficiently captures groups of linked sites that are
relevant to a trait or feature of interest.</p>
<p>There is no universally ‘correct’ way to delimit your region of
interest, it should instead be decided upon after carefully considering
your dataset, the purpose of your analysis, and the genomic features
surrounding a prospective site. Here, we’ll have a look at a few
possible methods that could be used to delimit a region of interest.</p>
</div>
<div class="section level2">
<h2 id="method-1-range-of-significant-snps-from-gwas">Method 1: Range of significant SNPs from GWAS<a class="anchor" aria-label="anchor" href="#method-1-range-of-significant-snps-from-gwas"></a>
</h2>
<p>Trait-associated regions identified through GWAS are particularly
suitable for analysis with crosshap as the phenotypic data used as input
can be directly used as input for crosshap to aid in capturing and
characterizing informative haplotypes. However, crosshap characterises a
specific region of interest, where GWAS only provides individual trait
associated loci.</p>
<p>General (GLM) and mixed linear (MLM) GWAS models provide association
results for all genomic sites, which allows the user to delimit a
genomic region by including all variants between the most distal loci
possessing a minimum level of significance. If your GWAS method combines
linked SNPs into single loci (e.g. <a href="https://doi.org/10.1371/journal.pgen.1005767" class="external-link">FarmCPU</a>), then
you can disentangle this by calculating pairwise linkage between all
SNPs in a wider region and delimit it based on the furthest loci above a
given threshold (e.g. R^2 with GWAS SNP &gt; 0.8).</p>
<p><img src="https://github.com/JacobIMarsh/crosshap/blob/main/images/Method1.jpeg?raw=true" width="650"></p>
<p>This method will result in a region that contains all highly
significant SNPs that are highly associated with a given trait.</p>
</div>
<div class="section level2">
<h2 id="method-2-haplotype-block-boundaries">Method 2: Haplotype block boundaries<a class="anchor" aria-label="anchor" href="#method-2-haplotype-block-boundaries"></a>
</h2>
<p>Sometimes you may be more interested in analysing the linkage
structures between variants agnostic of trait association information.
In these cases a visualized LD matrix (e.g.  using <a href="https://doi.org/10.1093/bib/bbaa227" class="external-link">LDBlockShow</a>) or the
boundaries of a haplotype block (e.g. defined using<br><a href="https://doi.org/10.1093/bioinformatics/bth457" class="external-link">Haploview</a>)
may be useful for delimiting a genomic region of interest.</p>
<p><img src="https://github.com/JacobIMarsh/crosshap/blob/main/images/Method2.jpeg?raw=true" width="650"></p>
<p>This method will result in a region with high homogeneous internal
linkage relative to distal sites and will ensure that you’re haplotyping
a region with strong linkage patterns.</p>
</div>
<div class="section level2">
<h2 id="method-3-gene-cluster">Method 3: Gene cluster<a class="anchor" aria-label="anchor" href="#method-3-gene-cluster"></a>
</h2>
<p>The co-location of two or more gene structures or independent
variants of interest can be useful for delimiting a region to explore
linkage patterns that span them. If there is a gene cluster you’re
interested in characterising, the positions of these features can
directly be used as the boundaries for your region of interest. It can
be useful to use this method alongside testing of general linkage in the
region (Method 2).</p>
<p><img src="https://github.com/JacobIMarsh/crosshap/blob/main/images/Method3.jpeg?raw=true" width="800"></p>
<p>This method will result in a region that contains relevant genomic
features, however this may not include with the breadth of linkage
patterns (i.e. Marker Groups may extend beyond the region analysed).</p>
</div>
<div class="section level2">
<h2 id="what-now">What now?<a class="anchor" aria-label="anchor" href="#what-now"></a>
</h2>
<p>After figuring out the boundaries of your region, the next step is to
subset your VCF to only include SNP information for that region.</p>
<p>To extract the region of interest from your VCF, you can use
something like tabix on the command line, just make sure you keep the
final header line that starts with ‘#CHR’:</p>
<p><code>$ grep -m 1 '#CHR' in.vcf &gt; in.region.vcf</code></p>
<p><code>$ bgzip in.vcf</code></p>
<p><code>$ tabix in.vcf.gz Chr1:100000-200000 &gt;&gt; in.region.vcf</code></p>
<p>Now your VCF is ready for analysis with crosshap! For more
information have a look at the <a href="Getting_started.html">Getting
started</a> vignette.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jacob Marsh.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
