<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="crosshap">
<title>Getting started • crosshap</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Getting started">
<meta property="og:description" content="crosshap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.2.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a>
    <a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a>
    <a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Getting started</h1>
                        <h4 data-toc-skip class="author">Jacob
Marsh</h4>
            
            <h4 data-toc-skip class="date">2023-03-05</h4>
      
      
      <div class="d-none name"><code>Getting_started.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>crosshap (R package) is an LD-based local haplotyping tool designed
to visualize patterns of variation across loci and individuals in a
genomic region of interest. In this tutorial we’ll walk through an
example analysis that will help to introduce and describe the main
functions of crosshap so you’ll be ready to jump into exploring your own
data.</p>
<p>The data we’ll be analyzing is soybean data from the USDA core
collection (Bayer et al. 2021). This tutorial will re-perform local
haplotype analysis of the seed protein QTL, cqProt-003, with crosshap
(Marsh et al. 2022). The data can be downloaded from the <a href="https://figshare.com/articles/dataset/fin_b51_173kb_only_vcf/22331167" class="external-link">crosshap
example data figshare repository</a>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jacobimarsh/crosshap"</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jacobimarsh.github.io/crosshap/">crosshap</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="preparing-inputs">Preparing inputs<a class="anchor" aria-label="anchor" href="#preparing-inputs"></a>
</h2>
<p>For all inputs a helper function is provided in crosshap to help read
them in consistently. For this tutorial, you’ll need to download the <a href="https://figshare.com/articles/dataset/fin_b51_173kb_only_vcf/22331167" class="external-link">crosshap
example data</a> and save it to a local directory.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Read example data into R and standardize using helper functions</span></span>
<span><span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_vcf.html">read_vcf</a></span><span class="op">(</span><span class="st">'path/to/fin_b51_173kb_only.vcf'</span><span class="op">)</span></span>
<span><span class="co">#VCF is provided as additional argument to help assign column and row names</span></span>
<span><span class="va">LD</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_LD.html">read_LD</a></span><span class="op">(</span><span class="st">'path/to/plink.ld'</span>, vcf <span class="op">=</span> <span class="va">vcf</span><span class="op">)</span></span>
<span><span class="va">pheno</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_pheno.html">read_pheno</a></span><span class="op">(</span><span class="st">'path/to/sprot_phen.txt'</span><span class="op">)</span></span>
<span><span class="va">metadata</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/read_metadata.html">read_metadata</a></span><span class="op">(</span><span class="st">'path/to/namepopfile.txt'</span><span class="op">)</span></span></code></pre></div>
<p><strong>VCF</strong> contains genomic variation (SNP) information of
985 accessions. Only sites within the 173kb trait associated region of
interest are present in the VCF, along with header information labeling
each individual.</p>
<details><summary><strong>How do I prepare my own VCF?</strong>
</summary><p>The data has been filtered and imputed; currently only biallelic
variant sites are supported by crosshap. We suggest you filter your data
prior to analysis, though imputation is optional and often it will be
preferable to keep it unimputed. For more information on how best to
isolate your region, take a look at vignette: <a href="Delimiting_region.html">Delimiting a region of interest</a>.</p>
</details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">vcf</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">11</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 11</span></span></span>
<span><span class="co">#&gt;   `#CHROM`    POS ID    REF   ALT   QUAL  FILTER INFO  FORMAT SRR1533182 USB.129</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> 20       3.16<span style="color: #949494;">e</span>7 20_3… C     A     .     PASS   .     GT     1|0        0|0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> 20       3.16<span style="color: #949494;">e</span>7 20_3… C     T     .     PASS   .     GT     0|0        0|0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> 20       3.16<span style="color: #949494;">e</span>7 20_3… A     G     .     PASS   .     GT     1|0        0|0    </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> 20       3.16<span style="color: #949494;">e</span>7 20_3… C     T     .     PASS   .     GT     0|0        0|0</span></span></code></pre></div>
<p><strong>LD matrix</strong> provides pairwise linkage information
between all SNP loci within the region of interest (173kb). Any LD
metric that captures pairwise correlation on a scale from 0 (unlinked)
to 1 (fully linked) may be used, including D’, R^2, or R^2 corrected for
kinship or population structure. For this example, we are using R^2
values.</p>
<details><summary><strong>How do I prepare my own LD matrix?</strong>
</summary><p>The LD file must be a square correlation matrix using the same loci
present in vcf, ordered by position, with or without position labels
(vcf object must be provided if position labels are missing). A simple
R^2 matrix can be calculated using PLINK (Purcell et al 2007) from which
the output can be directly used as input:</p>
<code>$ plink --r2 square --vcf imputed_173kb.vcf</code>
</details><div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">LD</span>, <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">4</span>,<span class="fl">4</span><span class="op">)</span><span class="op">)</span></span>
<span><span class="co">#&gt;             20_31604185 20_31604420 20_31604498 20_31604889</span></span>
<span><span class="co">#&gt; 20_31604185        1.00        0.00        0.24        0.96</span></span>
<span><span class="co">#&gt; 20_31604420        0.00        1.00        0.29        0.00</span></span>
<span><span class="co">#&gt; 20_31604498        0.24        0.29        1.00        0.23</span></span>
<span><span class="co">#&gt; 20_31604889        0.96        0.00        0.23        1.00</span></span></code></pre></div>
<p><strong>Phenotype</strong> scores for all, or at least some
individuals in the VCF is necessary for uncovering trait-associated
genomic variation and haplotypes. Given this tutorial analyzes a seed
protein QTL, seed protein % is the phenotype provided for this
example.</p>
<details><summary><strong>How do I prepare my own phenotype file?</strong>
</summary>
The phenotype file should be a comma or tab-delimited plain text file
with individual names (same as VCF IDs) in the first column, and
numerical phenotype scores in the second column, either with columns
untitled or labeled ‘Ind’ and ‘Pheno’. This can be the same phenotype
file provided for a GWAS.
</details><div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pheno</span>, n <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 2</span></span></span>
<span><span class="co">#&gt;   Ind   Pheno</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> AB.01  44  </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AB.02  50.5</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> BR.24  43.9</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> ESS    45.5</span></span></code></pre></div>
<p><strong>Metadata</strong> grouping labels for distinct subpopulations
in your dataset can aid in interpreting the haplotyping results. This
example provides level of domestication for each individual, including
undomesticated wild-type soy (wt), to domesticated landraces (lr),
domesticated cultivars improved by advanced breeding (ocult) and further
improved modern cultivars (mcult).</p>
<details><summary><strong>How do I prepare my own metadata file?</strong>
</summary>
The metadata file should be formatted the exact same as the phenotype
file, though with a categorical metadata grouping variable in the second
column instead of numerical phenotype score. This variable can be
anything you want to check the spread of across the haplotypes
(e.g. geographic region), or that could be associated with phenotypic
variation (e.g. year of collection). Metadata is an optional variable,
though is highly recommended to include.
</details><div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">metadata</span>, n <span class="op">=</span> <span class="fl">4</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4 × 2</span></span></span>
<span><span class="co">#&gt;   Ind   Metadata</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span> <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> AB.01 lr      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> AB.02 lr      </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> BR.01 mcult   </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> BR.02 mcult</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="local-haplotyping">Local haplotyping<a class="anchor" aria-label="anchor" href="#local-haplotyping"></a>
</h2>
<p>Local haplotyping involves classifying individuals by shared
haplotypes (combinations of alleles) across loci within a region of
interest. In crosshap, haplotypes are not defined by individual loci but
rather by allelic combinations of Marker Groups. Marker Groups are
clusters of tightly linked SNPs which are typically inherited as a unit
throughout the population. For an explanatory visualization showing how
Marker Group SNPs inform haplotyping with crosshap, take a look at our
vignette: <a href="hap_UMAP.html">Haplotype UMAP GIF</a>. The clustering
algorithm used to capture Marker Groups is DBSCAN (Ester et al. 1996)
which has two main parameters in crosshap: MGmin and epsilon.</p>
<p><strong>MGmin</strong> (minPts) is the minimum number of linked SNPs
in a cluster for it to be classified as a Marker Group instead of
background noise. In this example (2041 SNPs in 173kb region), a MGmin
of 30 was chosen.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Add minimum Marker Group SNP count</span></span>
<span><span class="va">MGmin</span> <span class="op">&lt;-</span> <span class="fl">30</span></span></code></pre></div>
<p><strong>Epsilon</strong> roughly translates to cluster density, when
epsilon increases, more SNPs will typically be included in more Marker
Groups. At this stage it’s worth providing a range of epsilons so you
can pick the best one later. This dataset has a lot of SNPs (2041 loci),
so the range of epsilons we will test will be quite low (0.2 - 1).</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Add list of epsilon values to run haplotyping on</span></span>
<span><span class="va">epsilon</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">0.2</span>,<span class="fl">0.4</span>,<span class="fl">0.6</span>,<span class="fl">0.8</span>,<span class="fl">1</span><span class="op">)</span></span></code></pre></div>
<p>Now you’ve got all your inputs and parameters ready, it’s time to run
the haplotyping!</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Run the haplotyping at all provided epsilon values</span></span>
<span><span class="va">HapObject</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/run_haplotyping.html">run_haplotyping</a></span><span class="op">(</span>vcf <span class="op">=</span> <span class="va">vcf</span>,</span>
<span>                LD <span class="op">=</span> <span class="va">LD</span>,</span>
<span>                pheno <span class="op">=</span> <span class="va">pheno</span>,</span>
<span>                metadata <span class="op">=</span> <span class="va">metadata</span>,</span>
<span>                epsilon <span class="op">=</span> <span class="va">epsilon</span>,</span>
<span>                MGmin <span class="op">=</span> <span class="va">MGmin</span><span class="op">)</span></span></code></pre></div>
<p>If everything went to plan, a haplotype object should have been saved
in <code>HapObject</code><br>
with haplotyping results for all five provided epsilon values. This
haplotype object will be needed to build the crosshap visualizations
(along with a chosen epsilon to visualize!).</p>
<details><summary><strong>How should I tune MGmin and epsilon for my dataset?</strong>
</summary><p>The MGmin value should be set based on the size, density and linkage
decay of your dataset. To give some scale: A 300kb region with 3000 SNPs
may be appropriately captured with a MGmin in the range of 20-40,
depending on the desired granularity. In contrast, a 50kb region with
100 SNPs may work better with an MGmin of 5-15.</p>
Your dataset may require increasing the epsilon range (e.g. to 1 - 3),
as especially for populations with lower linkage, too few Marker Groups
may be identified at the default range (0.2 - 1). If the Epsilon values
are raised very high, clusters will begin merging and the number of
distinct Marker Groups will decrease.
</details>
</div>
<div class="section level2">
<h2 id="optimizing-the-clustering-resolution">Optimizing the clustering resolution<a class="anchor" aria-label="anchor" href="#optimizing-the-clustering-resolution"></a>
</h2>
<p>Epsilon is a notoriously difficult parameter to optimize for DBSCAN,
the tool used to cluster SNPs into Marker Groups for haplotyping with
crosshap. Thankfully, crosshap has a clustree (Zappia et al 2018)
wrapper that summarizes the differences between the haplotype objects
created at the different epsilon values to help you pick the best
one.</p>
<p>The first clustering tree option provides a summary of changes in
haplotype groups resolved at different epsilon values.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Provide phenotype data and parameters used to create haplotype objects</span></span>
<span><span class="co">#Add type = 'MG' to ensure it summarizes Marker Groups rather than haplotypes</span></span>
<span><span class="va">hap_clustree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clustree_viz.html">clustree_viz</a></span><span class="op">(</span>HapObject <span class="op">=</span> <span class="va">HapObject</span>,</span>
<span>                              type <span class="op">=</span> <span class="st">'hap'</span><span class="op">)</span></span>
<span><span class="va">hap_clustree</span></span></code></pre></div>
<p><img src="Getting_started_files/figure-html/clustree-1.png" width="1152"></p>
<p>#<code>{r clustree, fig.width = 12, fig.height = 6} #Phenotype data and parameters used to create haplotype objects #Add type = 'hap' to ensure it summarizes haplotypes rather than Marker Groups hap_clustree &lt;- clustree_viz(HapObject,                               type = 'hap') hap_clustree #</code></p>
<details><summary><strong>What are the nodes and arrows on this plot?</strong>
</summary><p>Each row of nodes represents the different haplotypes (A-Z) idenfied
at a specific epsilon resolution (reported on right). The size of each
node indicates the frequency of individuals that possess a given
haplotype and the colour reflects the mean phenotype score for that
population of individuals. The ‘0’ node on the left represents the
population of individuals that are not placed in a haplotype. The arrows
between each row indicates that individuals in the upper haplotype node
are present in the lower haplotype node, with the opacity reflecting the
proportion of individuals.</p>
</details><p>What you may notice is that the number of haplotypes identified
changes from three populations at epsilon = 0.2 to six populations at
epsilon = 0.6, though remains constant with six haplotype populations as
epsilon increases to 1. This is a good sign that the haplotypes
identified at this epsilon range (0.6 - 1) are robust and effectively
capture a distinct genetic structure across the population in our region
of interest.</p>
<p>In addition, a range of colours are present for the rows of nodes
corresponding to epsilons 0.6 - 1 which indicates there’s phenotypic
diversity between a large haplotype A which on average has a low
phenotype score (seed protein %), and smaller haplotypes such as E and F
which on average have higher phenotype scores.</p>
<p>Given these factors, lets pick the smallest ‘stable’ epsilon value,
0.6, to investigate further with the full crosshap visualization.</p>
<details><summary><strong>What if I want to visualize changes between marker
groups?</strong>
</summary><p>Some analyses will be more interested in the linked SNPs which may be
causal for phenotypic trait variation, or may be markers for causal
variants. In this case, picking a clustering resolution (epsilon) that
is optimal for capturing informative Marker Groups is the goal. The
clustree wrapper provided by crosshap has a MG option for capturing
changes between Marker Groups, rather than haplotypes, between results
using different epsilon values. When <code>type = 'MG'</code>, the nodes
will instead represent Marker Groups, with size indicating number of
SNPs, and colour reflecting the phenotypic association of a given Marker
Groups. The ‘0’ group represents background SNPs that aren’t assigned to
a Marker Group.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Provide phenotype data and parameters used to create haplotype objects</span></span>
<span><span class="co">#Add type = 'MG' to ensure it summarizes Marker Groups rather than haplotypes</span></span>
<span><span class="va">MG_clustree</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/clustree_viz.html">clustree_viz</a></span><span class="op">(</span>HapObject <span class="op">=</span> <span class="va">HapObject</span>,</span>
<span>                              type <span class="op">=</span> <span class="st">'MG'</span><span class="op">)</span></span>
<span><span class="va">MG_clustree</span></span></code></pre></div>
<img src="Getting_started_files/figure-html/MG_clustree-1.png" width="1152"></details>
</div>
<div class="section level2">
<h2 id="visualizing-local-haplotypes">Visualizing local haplotypes<a class="anchor" aria-label="anchor" href="#visualizing-local-haplotypes"></a>
</h2>
<p>Now that we have our results ready as a haplotype object, and have
picked the optimal epsilon resolution for our dataset, it’s time to
visualize our results in full so we can start uncovering patterns of
variation!</p>
<p>On the crosshap visualization, the central matrix displays the
combinations of Marker Group alleles that define each haplotype, it
provides a grid around which features of each Marker Group and haplotype
combination is characterized. Features of haplotypes and the individuals
assigned to them are visualized vertically, with features in the top and
bottom plots. Features of Marker Groups and the SNPs assigned to them
are visualized horizontally, with features on the left and right
plots.</p>
<details><summary><strong>Why visualize Marker Groups alongside haplotypes?</strong>
</summary><p>The idea of the eponymous crosshap visualization is to integrate
Marker Group results (cluster of SNPs) with features of haplotype
populations (groups of genetically similar individuals), allowing you to
intuitively connect genetic variants with the individuals that possess
them. One of the reasons it’s important to capture both in the same
figure is because often trends of genetic variation captured by Marker
Groups can be explained by population structure across the haplotypes -
and vice versa.</p>
<p><br></p>
</details><details><summary><strong>What exactly is shown in each section of the
visualization?</strong>
</summary><ul>
<li><p>Above the central matrix is the frequency of individuals with
each haplotype combination, with a breakdown by our provided metadata
variable (level of domestication).</p></li>
<li><p>Below the central matrix is plotted the phenotype scores (seed
protein %) for each individual (each dot is an individual), grouped by
haplotype and coloured by metadata variable, like the top bar
plot.</p></li>
<li><p>Left of the central matrix is a summary of allele frequency
information for each of the Marker Groups (<strong>MG</strong>). This
indicates the frequency of minor allele, missing alleles, heterozygotes,
and reference alleles averaged across all the SNPs within each Marker
Group.</p></li>
<li><p>Right of the central matrix is plotted the phenotypic association
for each SNP (each dot is a SNP), grouped by Marker Group. The ‘pheno
association’ is effectively the difference in phenotype scores between
individuals with the alternate compared to reference alleles for a given
SNP.</p></li>
<li><p>The table in the bottom right summarizes the haplotype
information and the table in the top right summarizes the Marker Group
information. All guides are shown in the bottom left.</p></li>
</ul></details><div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Visualize haplotype object created by run_haplotyping() at the chosen epsilon (0.6)</span></span>
<span><span class="va">Hap_viz</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/crosshap_viz.html">crosshap_viz</a></span><span class="op">(</span>HapObject <span class="op">=</span> <span class="va">HapObject</span>, epsilon <span class="op">=</span> <span class="fl">0.6</span><span class="op">)</span></span>
<span><span class="va">Hap_viz</span></span></code></pre></div>
<p><img src="Getting_started_files/figure-html/crosshap_viz-1.png" width="1056"></p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#If it doesn't scale well in your plot viewer, try exporting it as an image with ggsave()</span></span></code></pre></div>
<details><summary><strong>Are there other options for these plots?</strong>
</summary>
The left plot providing allele frequency information can be swapped out
by providing the argument <code>plot_left = 'pos'</code> to instead
report the chromosomal position of each SNP in the region of interest.
Similarly. the right plot providing phenotypic association results for
SNPs can be swapped using <code>plot_right = 'cluster'</code> to instead
build a boxplot which indicates the level of internal linkage within
each Marker Group.
</details>
</div>
<div class="section level2">
<h2 id="interpreting-the-crosshap-visualization">Interpreting the crosshap visualization<a class="anchor" aria-label="anchor" href="#interpreting-the-crosshap-visualization"></a>
</h2>
<p>Alright, lets use the integrated visualization to uncover some
biological findings to put in that paper/report/grant application. Give
it a crack yourself, and if you get stuck, have a look at the example
findings below.</p>
<details><summary><strong>Level of domestication across haplotypes</strong>
</summary><p>Looking at the top bar plot and the summary table in the bottom
right, haplotype <strong>A</strong> almost exclusively contains
domesticated individuals including the vast majority of old and modern
improved cultivars. Haplotypes <strong>B</strong>, <strong>D</strong>
and <strong>E</strong> primarily contain wild-type individuals and
haplotypes <strong>C</strong> and <strong>F</strong> appear
landrace-specific.</p>
<p><br></p>
</details><details><summary><strong>Phenotypic variability across haplotypes</strong>
</summary><p>Looking at the bottom phenotype plot, individuals in haplotypes
<strong>B</strong>-<strong>F</strong> (which share the alternate allele
for <strong>MG1</strong>) typically have higher seed protein scores than
individuals in haplotype <strong>A</strong>. This trend is not only
found in the wild-type specific haplotypes
(<strong>B</strong>,<strong>D</strong>,<strong>E</strong>) but also the
landrace-specific haplotypes <strong>C</strong> and <strong>F</strong>
indicating there are some domesticated subpopulations that may share a
causal high-protein mutation in the region.</p>
<p><br></p>
</details><details><summary><strong>Linkage patterns between Marker Groups</strong>
</summary><p>By looking at the central MG/hap matrix, we can see that all other
Marker Groups have ‘nested’ linkage to <strong>MG1</strong>, meaning the
alternate alleles for <strong>MG2</strong>-<strong>MG9</strong> SNPs
tend to only be present when the alternate allele for
<strong>MG1</strong> is present. This may indicate a bottleneck, in this
case due to artificial selection for a variant tightly linked to
<strong>MG1</strong>, leading to dramatically reduced genetic variation
in the major domesticated haplotype <strong>A</strong>.</p>
<p><br></p>
</details><details><summary><strong>Phenotypic association of Marker Groups</strong>
</summary><p>Looking at the pheno association plot on the right, several Marker
Groups appear to be trait-associated, particularly <strong>MG1</strong>
and <strong>MG8</strong>. While <strong>MG8</strong> is associated with
the highest phenotypic effect (4.39), it has a low minor allele
frequency (0.03; see left allele count plot). In contrast,
<strong>MG1</strong> has a substantially higher minor allele frequency
(0.21), which means the phenotypic association for <strong>MG1</strong>
has more statistical power.</p>
<p><br></p>
</details><details><summary><strong>Candidates for further analysis and validation</strong>
</summary><p><strong>MG1</strong> is the strongest candidate Marker Groups for
further analysis, it may be worth investigating the functional
annotations of these markers to identify a candidate causal variant, or
testing whether indels/structural variation is linked to
<strong>MG1</strong>. <strong>MG8</strong> is also a candidate for
further analysis, though the alternate allele for <strong>MG8</strong>
is only found in 18 landrace individuals with very high seed protein
scores. As a result, it may be fruitful to investigate the subpopulation
of individuals in haplotype <strong>F</strong>: are they from the same
geographic region? do they share any other unique biological features?
These rare landrace individuals in haplotype <strong>F</strong> may be
interesting germplasm for introgression studies.</p>
<p><br></p>
</details><p>Congratulations, you did your first crosshap analysis! Now you’re
ready and equipped to give it a go with your data :)</p>
</div>
<div class="section level2">
<h2 id="exporting-raw-results-and-individual-plots">Exporting raw results and individual plots<a class="anchor" aria-label="anchor" href="#exporting-raw-results-and-individual-plots"></a>
</h2>
<p>You can have a closer look at the saved results by looking at the
elements within the haplotype object. This is ideal for exporting
information about the individuals that possess each haplotype, and the
SNPs within each Marker Group for further analysis.</p>
<ul>
<li>The Indfile reports the haplotype assigned to each individual</li>
<li>The Hapfile reports the identified haplotypes, and their Marker
Group combinations</li>
<li>The Varfile reports information for each SNP, including Marker Group
assignments</li>
</ul>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Print first lines of haplotype object buckets for epsilon = 0.6 results.</span></span>
<span><span class="va">HapObject</span><span class="op">$</span><span class="va">Haplotypes_MGmin30_E0.6</span><span class="op">$</span><span class="va">Indfile</span></span>
<span><span class="va">HapObject</span><span class="op">$</span><span class="va">Haplotypes_MGmin30_E0.6</span><span class="op">$</span><span class="va">Hapfile</span></span>
<span><span class="va">HapObject</span><span class="op">$</span><span class="va">Haplotypes_MGmin30_E0.6</span><span class="op">$</span><span class="va">Varfile</span></span></code></pre></div>
<p>If you’re interested in a specific plot from the crosshap
visualization, though don’t need the full integrated figure, you can
build the plots one-by one and stitch them together as you like using
the patchwork package in R (Pederson 2022).</p>
<p>Below we’ll only include the top and bottom plots that are relevant
that characterize our haplotype populations.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Install patchwork if you don't have it using install.packages("patchwork")</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://patchwork.data-imaginist.com" class="external-link">patchwork</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#First lets build the top plot, keeping the guides with "hide_labels = F"</span></span>
<span><span class="va">top</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_top_metaplot.html">build_top_metaplot</a></span><span class="op">(</span>HapObject <span class="op">=</span> <span class="va">HapObject</span>, epsilon <span class="op">=</span> <span class="fl">0.6</span>, hide_labels <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Next lets build the bottom plot, hiding the guides with "hide_labels = T"</span></span>
<span><span class="va">bot</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/build_bot_halfeyeplot.html">build_bot_halfeyeplot</a></span><span class="op">(</span>HapObject <span class="op">=</span> <span class="va">HapObject</span>, epsilon <span class="op">=</span> <span class="fl">0.6</span>, hide_labels <span class="op">=</span> <span class="cn">T</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Now we can stitch them together with patchwork!</span></span>
<span><span class="va">hap_plots</span> <span class="op">&lt;-</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/wrap_plots.html" class="external-link">wrap_plots</a></span><span class="op">(</span><span class="va">top</span>, <span class="va">bot</span><span class="op">)</span> <span class="op">+</span> <span class="fu">patchwork</span><span class="fu">::</span><span class="fu"><a href="https://patchwork.data-imaginist.com/reference/plot_layout.html" class="external-link">plot_layout</a></span><span class="op">(</span>ncol <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span><span class="va">hap_plots</span></span></code></pre></div>
<p><img src="Getting_started_files/figure-html/export_subplots-1.png" width="700"></p>
</div>
<div class="section level2">
<h2 id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<p>BAYER, P. E., VALLIYODAN, B., HU, H., MARSH, J. I., YUAN, Y., VUONG,
T. D., PATIL, G., SONG, Q., BATLEY, J., VARSHNEY, R. K., LAM, H. M.,
EDWARDS, D. &amp; NGUYEN, H. T. 2021. Sequencing the USDA core soybean
collection reveals gene loss during domestication and<br>
breeding. <em>Plant Genome</em>, e20109.</p>
<p>MARSH, J. I., HU, H., PETEREIT, J., BAYER, P. E., VALLIYODAN, B.,
BATLEY, J., NGUYEN, H. T. &amp; EDWARDS, D. 2022. Haplotype mapping
uncovers unexplored variation in wild and domesticated soybean at the
major protein locus <em>cqProt-003</em>. <em>Theor Appl Genet</em>, 135,
1443-1455.</p>
<p>PURCELL, S., NEALE, B., TODD-BROWN, K., THOMAS, L., FERREIRA, M. A.
R., BENDER, D., MALLER, J., SKLAR, P., DE BAKKER, P. I. W., DALY, M. J.
&amp; SHAM, P. C. 2007. PLINK: A tool set for whole-genome association
and population-based linkage analyses. <em>Am J Hum Genet</em>, 81,
559-575.</p>
<p>ESTER, M., KRIEGEL, H.-P., SANDER, J. &amp; XU, X. A density-based
algorithm for discovering clusters in large spatial databases with
noise. Proceedings of the 2nd ACM International Conference on Knowledge
Discovery and Data Mining (KDD), 1996. 226-231.</p>
<p>PEDERSON, T. 2022. Patchwork: The composer of plots.<br><a href="https://patchwork.data-imaginist.com" class="external-link uri">https://patchwork.data-imaginist.com</a>, <a href="https://github.com/thomasp85/patchwork" class="external-link uri">https://github.com/thomasp85/patchwork</a>.</p>
<p>ZAPPIA, L. &amp; OSHLACK, A. 2018. Clustering trees: A visualization
for evaluating clusterings at multiple resolutions.
<em>Gigascience</em>, 7.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jacob Marsh, Brady Johnston, Jakob Petereit.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
