<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="crosshap">
<title>Haplotype UMAP GIF • crosshap</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Haplotype UMAP GIF">
<meta property="og:description" content="crosshap">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">crosshap</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.1.0</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/Delimiting_region.html">Delimiting a region of interest</a>
    <a class="dropdown-item" href="../articles/Getting_started.html">Getting started</a>
    <a class="dropdown-item" href="../articles/hap_UMAP.html">Haplotype UMAP GIF</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="" class="logo" alt=""><h1>Haplotype UMAP GIF</h1>
                        <h4 data-toc-skip class="author">Jacob
Marsh</h4>
            
            <h4 data-toc-skip class="date">2023-03-10</h4>
      
      
      <div class="d-none name"><code>hap_UMAP.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>Between haplotypes, Marker Groups, allelic combinations and unplaced
loci, there’s a lot of data transformation that crosshap performs to the
point where it can be difficult to take a step back and appreciate what
genomic patterns are shared between individuals with the same assigned
haplotype. Here, we’ll make an animated GIF that will help us visualize
this information. The GIF doesn’t necessarily reveal more details about
our haplotypes, however it can be useful as a sanity check, and to get a
better understanding of what our results reflect.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Load relevant libraries</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jacobimarsh.github.io/crosshap/">crosshap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/tkonopka/umap" class="external-link">umap</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://gganimate.com" class="external-link">gganimate</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tibble.tidyverse.org/" class="external-link">tibble</a></span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Read in example data (LD Matrix, VCF, Haplotype object and chosen epsilon)</span></span>
<span><span class="va">LD</span> <span class="op">&lt;-</span> <span class="fu">crosshap</span><span class="fu">::</span><span class="va"><a href="../reference/LD.html">LD</a></span></span>
<span><span class="va">vcf</span> <span class="op">&lt;-</span> <span class="fu">crosshap</span><span class="fu">::</span><span class="va"><a href="../reference/vcf.html">vcf</a></span></span>
<span><span class="va">HapObject</span> <span class="op">&lt;-</span> <span class="fu">crosshap</span><span class="fu">::</span><span class="va"><a href="../reference/HapObject.html">HapObject</a></span></span>
<span><span class="va">epsilon</span> <span class="op">&lt;-</span> <span class="fl">0.6</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="umap">UMAP<a class="anchor" aria-label="anchor" href="#umap"></a>
</h2>
<p>Uniform Manifold Approximation Projection (<a href="https://doi.org/10.48550/arXiv.1802.03426" class="external-link">UMAP</a>) is a useful
dimension reduction tool for visualizing clustering results. In brief,
it will attempt to co-locate points (SNPs) that are correlated (in
linkage) in two-dimensional space, similar to PCA.</p>
<p>You’ll need to set several UMAP parameters. The data input for UMAP
can thankfully be taken as the LD matrix provided to crosshap for
haplotyping. <code>n_neighbors</code> should match the MGmin parameter
provided to crosshap for haplotyping. The final two parameters,
<code>min_dist</code> and <code>spread</code> will need some testing to
optimize.</p>
</div>
<div class="section level2">
<h2 id="capturing-marker-groups-with-umap">Capturing Marker Groups with UMAP<a class="anchor" aria-label="anchor" href="#capturing-marker-groups-with-umap"></a>
</h2>
<p>First lets visualize where our Marker Groups are in the UMAP which
can be a useful reference for later, and can help us optimize our UMAP
before we use it to map our individuals.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co">#Run UMAP on LD matrix to get x &amp; y coordinates for plotting</span></span>
<span><span class="va">umap_in</span> <span class="op">&lt;-</span> <span class="fu">umap</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/umap/man/umap.html" class="external-link">umap</a></span><span class="op">(</span><span class="va">LD</span>, min_dist <span class="op">=</span> <span class="fl">2</span>, spread <span class="op">=</span> <span class="fl">2.5</span>, n_neighbors <span class="op">=</span> <span class="fl">30</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Add UMAP coordinates to SNPs with Marker Group information</span></span>
<span><span class="va">MGumap_data</span> <span class="op">&lt;-</span> <span class="va">umap_in</span><span class="op">$</span><span class="va">layout</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/cbind.html" class="external-link">cbind</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/colnames.html" class="external-link">rownames</a></span><span class="op">(</span><span class="va">umap_in</span><span class="op">$</span><span class="va">data</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> <span class="fu">tibble</span><span class="fu">::</span><span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/rename.html" class="external-link">rename</a></span><span class="op">(</span>UMAP1 <span class="op">=</span> <span class="va">V1</span>, UMAP2 <span class="op">=</span> <span class="va">V2</span>, ID <span class="op">=</span> <span class="st">'rownames(umap_in$data)'</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span><span class="co">#HapObject$Haplotypes_MGmin30_E0.6$Varfile contains variant information at epsilon = 0.6</span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">left_join</a></span><span class="op">(</span><span class="va">HapObject</span><span class="op">$</span><span class="va">Haplotypes_MGmin30_E0.6</span><span class="op">$</span><span class="va">Varfile</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span> </span>
<span>                     <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="va">ID</span>, <span class="va">MGs</span><span class="op">)</span>, by <span class="op">=</span> <span class="st">"ID"</span><span class="op">)</span> <span class="op"><a href="../reference/pipe.html">%&gt;%</a></span></span>
<span>  <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>MGs <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/grep.html" class="external-link">gsub</a></span><span class="op">(</span><span class="st">'0'</span>, <span class="cn">NA</span>, <span class="va">MGs</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="co">#Plot SNPs by UMAP coordinates, coloured by Marker Group </span></span>
<span><span class="va">MGumap</span> <span class="op">&lt;-</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">MGumap_data</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">UMAP1</span>, y <span class="op">=</span> <span class="va">UMAP2</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.4</span>, <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="va">MGs</span><span class="op">)</span>, size <span class="op">=</span> <span class="fl">1</span>, na.rm <span class="op">=</span> <span class="cn">F</span><span class="op">)</span><span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guides.html" class="external-link">guides</a></span><span class="op">(</span>colour <span class="op">=</span> <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/guide_legend.html" class="external-link">guide_legend</a></span><span class="op">(</span>override.aes <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span>size <span class="op">=</span> <span class="fl">5</span>, alpha <span class="op">=</span> <span class="fl">0.7</span><span class="op">)</span>, title <span class="op">=</span> <span class="st">"Marker group"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">MGumap</span></span></code></pre></div>
<p><img src="https://github.com/JacobIMarsh/crosshapimages/blob/main/images/MGumap.png?raw=true" width="500"></p>
<details><summary><strong>What haplotypes do these Marker Groups define?</strong>
</summary><div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu">crosshap</span><span class="fu">::</span><span class="fu"><a href="../reference/build_mid_dotplot.html">build_mid_dotplot</a></span><span class="op">(</span><span class="va">HapObject</span>, epsilon <span class="op">=</span> <span class="fl">0.6</span>, hide_labels <span class="op">=</span> <span class="cn">F</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/JacobIMarsh/crosshapimages/blob/main/images/middotplot.png?raw=true" width="500"></p>
</details><p>A full guide to optimizing UMAP parameters is beyond the scope of
this tutorial, but the MGumap plot we’ve created should be able to help
with that if needed. The goal when optimizing your UMAP is to find
parameters that co-locate SNPs of the same Marker Group in distinct
clusters. Read <a href="https://umap-learn.readthedocs.io/en/latest/parameters.html#min-dist" class="external-link">here</a>
for more information on how to optimize <code>min_dist</code> and don’t
forget to keep <code>spread</code> above <code>min_dist()</code>!</p>
</div>
<div class="section level2">
<h2 id="building-umap-gif">Building UMAP GIF<a class="anchor" aria-label="anchor" href="#building-umap-gif"></a>
</h2>
<p>The <code><a href="../reference/prepare_hap_umap.html">prepare_hap_umap()</a></code> function in crosshap takes UMAP
results and a haplotype object built by <code><a href="../reference/run_haplotyping.html">run_haplotyping()</a></code>
with an epsilon resolution and builds a plot for individuals from each
haplotype with their SNP alleles across the UMAP.</p>
<details><summary><strong>What’s the <code>nsamples</code> parameter?</strong>
</summary><p>The <code>nsamples</code> parameter will match the number of unique
frames in your final GIF. It can be changed depending on the number of
randomly sampled individuals you’d like to have from each haplotype. 25
is the default, however if you have more individuals it may be worth
increasing, or if it is taking too long to run consider lowering this
value. NOTE: nsamples can include the same individual multiple times
which is useful when nsamples &gt; nIndividuals within a given
haplotype.</p>
</details><div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pre_anim_gg</span> <span class="op">&lt;-</span> <span class="fu">crosshap</span><span class="fu">::</span><span class="fu"><a href="../reference/prepare_hap_umap.html">prepare_hap_umap</a></span><span class="op">(</span><span class="va">umap_in</span>,</span>
<span>                            HapObject <span class="op">=</span> <span class="va">HapObject</span>,</span>
<span>                            epsilon <span class="op">=</span> <span class="fl">0.6</span>,</span>
<span>                            vcf <span class="op">=</span> <span class="va">vcf</span>,</span>
<span>                            nsamples <span class="op">=</span> <span class="fl">25</span><span class="op">)</span></span></code></pre></div>
<p>At this stage, plots for all frames for all the individuals from all
haplotypes are stacked onto a single image, so we’ll need to isolate
them. First, individuals are separated into different facets by
haplotype. Then, gganimate will stagger the images by only showing one
individual from each haplotype at a time. Finally, the animation will be
rendered with the <code><a href="https://gganimate.com/reference/renderers.html" class="external-link">gifski_renderer()</a></code>.</p>
<p>Feel free to experiment with the <code>transition_length</code>,
<code>state_length</code> and <code><a href="https://gganimate.com/reference/animate.html" class="external-link">animate()</a></code> display parameters
to find what you think best represents your data.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">hap_gganim</span> <span class="op">&lt;-</span> <span class="va">pre_anim_gg</span> <span class="op">+</span></span>
<span>  <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">hap</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu">gganimate</span><span class="fu">::</span><span class="fu"><a href="https://gganimate.com/reference/transition_states.html" class="external-link">transition_states</a></span><span class="op">(</span><span class="va">Frame</span>,</span>
<span>                    transition_length <span class="op">=</span> <span class="fl">0</span>,</span>
<span>                    state_length <span class="op">=</span> <span class="fl">1</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gganimate</span><span class="fu">::</span><span class="fu"><a href="https://gganimate.com/reference/animate.html" class="external-link">animate</a></span><span class="op">(</span></span>
<span>  <span class="va">hap_gganim</span>,</span>
<span>  renderer <span class="op">=</span> <span class="fu">gganimate</span><span class="fu">::</span><span class="fu"><a href="https://gganimate.com/reference/renderers.html" class="external-link">gifski_renderer</a></span><span class="op">(</span><span class="op">)</span>,</span>
<span>  fps <span class="op">=</span> <span class="fl">3</span>,</span>
<span>  res <span class="op">=</span> <span class="fl">1200</span>,</span>
<span>  width  <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  height <span class="op">=</span> <span class="fl">6</span>,</span>
<span>  units <span class="op">=</span> <span class="st">"in"</span>,</span>
<span>  <span class="co">#NOTE: nframes = nsamples</span></span>
<span>  nframes <span class="op">=</span> <span class="fl">25</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu">gganimate</span><span class="fu">::</span><span class="fu"><a href="https://gganimate.com/reference/anim_save.html" class="external-link">anim_save</a></span><span class="op">(</span><span class="st">"hap_animation.gif"</span><span class="op">)</span></span></code></pre></div>
<p><img src="https://github.com/JacobIMarsh/crosshapimages/blob/main/images/hap_animation.gif?raw=true" width="650"></p>
<details><summary><strong>What do the dots and facets indicate here?</strong>
</summary><p>In the GIF, the dots represent the same SNPs as in the MGumap,
however now each point is coloured only if the individual possesses the
alternate allele at that locus, with dark grey for SNPs unassigned to
Marker Groups and light grey for reference alleles. Each facet displays
individuals that share a given haplotype. The first facet shows all
Marker Group SNPs for reference, it’s the same as the MGumap we built
before. Individuals that are unassigned to a haplotype are placed in the
final ‘Unassigned’ facet.</p>
</details><p>What you should notice is that individuals within the same haplotype
will share the majority of alleles for Marker Group SNPs. However,
variability is present across ‘noise’ SNPs that are not assigned to a
marker group between individuals in the same haplotype group. In other
words, the colours within each facet stay the same, reflecting shared
haplotypes for MG SNPs, though background variation may differ.</p>
<p>Congratulations on building your very own haplotype UMAP GIF
visualization, go show your friends!</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jacob Marsh, Brady Johnston, Jakob Petereit.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
